met-web/src/components/engagement/form/EngagementWidgets/Events/VirtualSessionFormDrawer.tsx

```diff
@@ -0,0 +1,220 @@
+import React, { useContext, useState, useEffect } from 'react';
+import Box from '@mui/material/Box';
+import Drawer from '@mui/material/Drawer';
+import Divider from '@mui/material/Divider';
+import { Grid } from '@mui/material';
+import { MetHeader3, MetLabel, PrimaryButton, SecondaryButton } from 'components/common';
+import { useForm, FormProvider, SubmitHandler } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as yup from 'yup';
+import { useAppDispatch } from 'hooks';
+import { EventsContext } from './EventsContext';
+import ControlledTextField from 'components/common/ControlledInputComponents/ControlledTextField';
+import { openNotification } from 'services/notificationService/notificationSlice';
+import { postEvent } from 'services/widgetService/EventService';
+import { EVENT_TYPE } from 'models/event';
+import { formatToUTC } from 'components/common/dateHelper';
+import dayjs from 'dayjs';
+
+const schema = yup
+    .object({
+        description: yup.string().max(500, 'Description cannot exceed 500 characters'),
+        session_link: yup.string().required('Session link cannot be empty'),
+        session_link_text: yup.string().default('Click here to register').required('Session Link Text cannot be empty'),
+        date: yup.string().defined().required('Date cannot be empty'),
+        time_from: yup.string().required('Time from cannot be empty'),
+        time_to: yup.string().required('Time to cannot be empty'),
+    })
+    .required();
+
+type VirtualSessionForm = yup.TypeOf<typeof schema>;
+
+const VirtualSessionFormDrawer = () => {
+    const dispatch = useAppDispatch();
+    const { virtualSessionFormTabOpen, setVirtualSessionFormTabOpen, widget, loadEvents } = useContext(EventsContext);
+    const [isCreating, setIsCreating] = useState(false);
+
+    const methods = useForm<VirtualSessionForm>({
+        resolver: yupResolver(schema),
+    });
+
+    useEffect(() => {
+        methods.setValue('session_link_text', 'Click here to register');
+    }, []);
+
+    const { handleSubmit, reset } = methods;
+
+    const onSubmit: SubmitHandler<VirtualSessionForm> = async (data: VirtualSessionForm) => {
+        console.log('ON SUBMIT!!!!');
+        if (!widget) {
+            return;
+        }
+        const validatedData = await schema.validate(data);
+        try {
+            setIsCreating(true);
+            const { description, session_link, session_link_text, date, time_from, time_to } = validatedData;
+            const time_from_split = time_from.split(':');
+            const time_to_split = time_to.split(':');
```

Comment:
I had a mind to cleanup this section in the other drawer. Just to export them into a separate function called something like getDates that returns { dateFrom: ..., dateTo: ...}.

Reply:
function added to date helper

Reply:
Yeah I see it has been cleaned up in add in person event, could be used here too no?

Reply:
yeah, could we please add the time split there too?

Reply:
It's not a very common function, so I would create a a utils file for it under the Events folder. But okay to also keep it in common

Reply:
Added a utils file